package/boot/uboot-envtools/files/lantiq

```diff
@@ -3,32 +3,37 @@
 # Copyright (C) 2012 OpenWrt.org
 #
 
-[ -e /etc/config/ubootenv ] && exit 0
-
-touch /etc/config/ubootenv
-
 . /lib/uboot-envtools.sh
 . /lib/functions.sh
 
+idx="$(find_mtd_index uboot_env)"
+
+[ -n "$idx" ] || exit 0
+
+[ -e /etc/config/ubootenv -a "$(uci get ubootenv.@ubootenv[-1].dev)" = "/dev/mtd$idx" ] && exit 0
+
+rm -f /etc/config/ubootenv
+rm -f /etc/fw_env.config
+
+touch /etc/config/ubootenv
+
 board=$(board_name)
 
 case "$board" in
 bt,homehub-v2b)
-	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x10000" "0x10000" "1"
+	ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x10000" "0x10000" "1"
 	;;
 bt,homehub-v3a)
-	ubootenv_add_uci_config "/dev/mtd2" "0x0" "0x4000" "0x4000" "1"
+	ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x4000" "0x4000" "1"
 	;;
 siemens,gigaset-sx76x)
-	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x10000" "0x10000" "1"
+	ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x10000" "0x10000" "1"
```

Comment:
This can be combined with bt,homehub-v2b now.

Reply:
OK, I've combined that two.